src/m365/planner/commands/task/task-list.ts

```diff
@@ -14,20 +26,193 @@ class PlannerTaskListCommand extends GraphItemsListCommand<Task> {
   }
 
   public get description(): string {
-    return 'Lists Planner tasks for the currently logged in user';
+    return 'Lists the planner tasks in a bucket, plan, or tasks for the currently logged in user';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.bucketId = typeof args.options.bucketId !== 'undefined';
+    telemetryProps.bucketName = typeof args.options.bucketName !== 'undefined';
+    telemetryProps.planId = typeof args.options.planId !== 'undefined';
+    telemetryProps.planName = typeof args.options.planName !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
   }
 
   public defaultProperties(): string[] | undefined {
-    return ['id', 'title', 'startDateTime', 'dueDateTime', 'completedDateTime'];
+    return ['id', 'title', 'startDateTime', 'dueDateTime', 'completedDateTime', 'planId', 'bucketId'];
   }
 
   public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
-    this
-      .getAllItems(`${this.resource}/v1.0/me/planner/tasks`, logger, true)
-      .then((): void => {
-        logger.log(this.items);
-        cb();
-      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+    const bucketId: string | undefined = args.options.bucketId;
+    const bucketName: string | undefined = args.options.bucketName;
+    const planId: string | undefined = args.options.planId;
+    const planName: string | undefined = args.options.planName;
+
+    if (bucketId || bucketName) {
+      this
+        .getBucketId(args)
+        .then((bucketId: string): Promise<void> => this.getAllItems(`${this.resource}/v1.0/planner/buckets/${bucketId}/tasks`, logger, true))
+        .then((): void => {
+          logger.log(this.items);
+          cb();
+        }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+    }
+    else if (planId || planName) {
+      this
+        .getPlanId(args)
+        .then((planId: string): Promise<void> => this.getAllItems(`${this.resource}/v1.0/planner/plans/${planId}/tasks`, logger, true))
+        .then((): void => {
+          logger.log(this.items);
+          cb();
+        }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+    }
+    else {
+      this
+        .getAllItems(`${this.resource}/v1.0/me/planner/tasks`, logger, true)
+        .then((): void => {
+          logger.log(this.items);
+          cb();
+        }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+    }
+  }
+
+  private getBucketId(args: CommandArgs): Promise<string> {
+    if (args.options.bucketId) {
+      return Promise.resolve(args.options.bucketId);
```

Comment:
Since `bucketId` is provided by the user, we should encode it before returning it.